client/src/client.rs

```diff
@@ -605,7 +608,7 @@ impl Client {
             let mut event_iterator = {
                 let event_iterator_result = tokio::time::timeout_at(
                     deadline,
-                    self.listen_for_events_async(PipelineEventFilter::new().for_hash(hash.into())),
+                    self.listen_for_events_async(TransactionEventFilter::default().for_hash(hash)),
```

Comment:
@DCNick3 I'm hitting a wall here. I need to filter by both all block events and transaction events with a certain hash value. Should I change the API to take a list of filters or is there a way to express this with current filtering API?

Reply:
This is what #4356 is about, and having a list of filters is the approach I am gravitating towards. One concern with this approach is having to limit the `EventFilter` size in all the relevant places (in transactions and in querying requests). Not sure what the policy should be exactly though

Reply:
so then, this PR is ready for review IMO